substrate-node/pallets/pallet-tfgrid/src/lib.rs

```diff
@@ -2417,41 +2327,24 @@ impl<T: Config> Pallet<T> {
         Ok(name_parsed)
     }
 
-    fn get_public_ip_ip(
-        public_ip_ip: PublicIpIpInput,
-    ) -> Result<PublicIpIpOf<T>, DispatchErrorWithPostInfo> {
-        let public_ip_ip_parsed = <T as Config>::PublicIP::try_from(public_ip_ip)?;
-        Ok(public_ip_ip_parsed)
-    }
-
-    fn get_public_ip_gw(
-        public_ip_gw: PublicIpGatewayInput,
-    ) -> Result<PublicIpGatewayOf<T>, DispatchErrorWithPostInfo> {
-        let public_ip_gw_parsed = <T as Config>::GatewayIP::try_from(public_ip_gw)?;
-        Ok(public_ip_gw_parsed)
-    }
-
     fn get_public_ips(
         public_ips: PublicIpListInput<T>,
-    ) -> Result<PublicIpListOf<T>, DispatchErrorWithPostInfo> {
-        let mut public_ips_list: PublicIpListOf<T> = vec![].try_into().unwrap();
+    ) -> Result<PublicIpListOf, DispatchErrorWithPostInfo> {
+        let mut public_ips_list: PublicIpListOf = vec![].try_into().unwrap();
 
         for ip in public_ips {
-            let public_ip_ip = Self::get_public_ip_ip(ip.ip)?;
-            let public_ip_gateway = Self::get_public_ip_gw(ip.gw)?;
-
             if public_ips_list.contains(&PublicIP {
-                ip: public_ip_ip.clone(),
-                gateway: public_ip_gateway.clone(),
+                ip: ip.ip.clone(),
+                gateway: ip.gw.clone(),
                 contract_id: 0,
             }) {
                 return Err(DispatchErrorWithPostInfo::from(Error::<T>::IpExists));
             }
 
             public_ips_list
                 .try_push(PublicIP {
-                    ip: public_ip_ip,
-                    gateway: public_ip_gateway,
+                    ip: ip.ip,
+                    gateway: ip.gw,
                     contract_id: 0,
                 })
                 .or_else(|_| {
```

Comment:
The error below is actually bad since this error actually originates from the `try_push` on a `BoundedVec`, so an error here would mean said vec is over capacity, not that the config is invalid.

Additionally, it does not achieve anything, since the return is part of the closure defined, so it will just return said error out of the closure as the new Err value instead of the existing one.

Also, since you are just converting the Err type, `map_err` is a better function here to do so, as this better expresses the intent.

Lastly the `ok()` is pointless since it converts a Result into an Option but it is never assigned or otherwise handled. If you want to return the (mapped) error then replace `ok()` with `?`